### ISSUE TITLE
Add release builds for `aarch64-apple-darwin`

### ISSUE THREAD

**Issue by jszwedko** [vectordotdev/vector#10193]
A user tried to install Vector via Homebrew on an M1 Mac but discovered we don't publish binaries for this architecture.

**Comment by rdooley** [comment]
Tried via homebrew and [the quickstart curl script](https://vector.dev/docs/setup/quickstart/#install-vector)

**Comment by FlatMapIO** [comment]
`aarch64-apple-darwin` not found in https://packages.timber.io/vector/0.18.1

**Comment by spencergilbert** [comment]
> `aarch64-apple-darwin` not found in https://packages.timber.io/vector/0.18.1

Correct, we are not publishing binaries for that arch today, this issue exists to track introducing it. You should be able to build locally if you need to though.

**Comment by panophobicPanda** [comment]
I'm also interested in a build of vector for M1. 
I know there are no github runners for M1 yet but would it be possible to cross-compile for M1 on the mac x86_64 image?
I'm curious if the team plans to wait for M1 runners or if a different approach is planned.

**Comment by jszwedko** [comment]
Hi @panophobicPanda ! The framework we use for cross-compilation, `cross`, does not yet support `aarch64-apple-darwin` as a platform (https://github.com/cross-rs/cross/issues/508). We do use self-hosted runners in AWS for the Windows builds. I could see us doing the same for M1 Mac builds when they are [generally available](https://aws.amazon.com/ec2/instance-types/mac/).

**Comment by jszwedko** [comment]
Just a note for readers that it is possible to compile Vector natively on M1 Macs (`make build`).

**Comment by rdooley** [comment]
> Just a note for readers that it is possible to compile Vector natively on M1 Macs (`make build`).

This is what I've settled on for now and it works great, albeit not as slick.

**Comment by jszwedko** [comment]
👍 AWS also has new support for M1 runners that we may be able to leverage too.

**Comment by panophobicPanda** [comment]
Since a few things have changed, it seems like there are now a few different options here?

1. Build with `cross-rs`
2. Build on an M1 hosted in AWS (github runner now supports M1)
3. Build on a hosted github runner (doesn't exist yet)

I wanted to get an idea of timing... are we planning on waiting for #3 ?

**Comment by jszwedko** [comment]
> Since a few things have changed, it seems like there are now a few different options here?
> 
> 1. Build with `cross-rs`
> 2. Build on an M1 hosted in AWS (github runner now supports M1)
> 3. Build on a hosted github runner (doesn't exist yet)
> 
> I wanted to get an idea of timing... are we planning on waiting for #3 ?

We are currently waiting for #3 there, but if there is a way to get it working via cross I'd be interested to see that. We'd like to avoid needing to manage an M1 instance in AWS but if we get more demand for this, that would be another option.

**Comment by jszwedko** [comment]
> GitHub now offers macOS runners (since some time): https://github.blog/news-insights/product-news/introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/
> 
> Is there anything particular blocking adding Apple Silicon releases?

Indeed! Nothing blocking at the moment, I don't think, just time.

**Comment by jszwedko** [comment]
Just a heads up that the v0.44.0 release of Vector has `aarch64-apple-darwin` artifacts 🎉 We are still updating the release page to link to them, but you can find them [on GitHub](https://github.com/vectordotdev/vector/releases/tag/v0.44.0). The homebrew formula has been updated too.
